packages/flutter/lib/src/widgets/widget_state.dart

```diff
@@ -45,14 +45,77 @@ abstract interface class WidgetStatesConstraint {
   bool isSatisfiedBy(Set<WidgetState> states);
 }
 
-// A private class, used in [WidgetStateOperators].
-class _WidgetStateOperation implements WidgetStatesConstraint {
-  const _WidgetStateOperation(this._isSatisfiedBy);
+@immutable
+sealed class _WidgetStateCombo implements WidgetStatesConstraint {
+  const _WidgetStateCombo(this.first, this.second);
 
-  final bool Function(Set<WidgetState> states) _isSatisfiedBy;
+  final WidgetStatesConstraint first;
+  final WidgetStatesConstraint second;
 
   @override
-  bool isSatisfiedBy(Set<WidgetState> states) => _isSatisfiedBy(states);
+  // ignore: hash_and_equals, since == is defined in subclasses
```

Comment:
I get it why you want to "reuse" the `int get hashCode => Object.hash(first, second);` implementation. However, seeing the ignores feels a bit off to me personally.

When reading this code, one would probably wonder why we step away from the usual "hashCode + operator==" pattern. I did a quick search and the framework does not have any other `ignore: hash_and_equals` instances.

I would prefer to remove the ignore and define the `int get hashCode => Object.hash(first, second);` in the subclasses anyway. Then we don't break from the familiar pattern _and_ do not introduce an ignore.

Is just a preference, though.

Reply:
What if we have a private static method on the super class? Since this is a `sealed` hierarchy, it has to stay in the same library unit (= Dart file) anyway.

Reply:
I believe an instance method that calls a static method might have a performance impact, similar to https://github.com/flutter/flutter/issues/144156#issuecomment-1967607869

Reply:
I see both points here... I think with the little description `since == is defined in subclasses`, it's easy enough to look up what the == operation will be, so personally I don't mind keeping it as-is.